src/test/crossFetchNode.test.ts

```diff
@@ -274,3 +274,105 @@ test('should not fail when arbitrary parameters are configured in fetchOptions',
   });
   expect(response).toEqual({});
 });
+
+test('throws on error responses', async () => {
+  nock('https://localhost:3000')
+    .get(
+      `/search/shopper-search/v1/organizations/${config.parameters.organizationId}/product-search?siteId=${config.parameters.siteId}&q=sony`
+    )
+    .matchHeader('authorization', 'Bearer test-auth')
+    .reply(
+      400,
+      {content: 'not empty'},
+      {'content-type': 'application-json charset=UTF-8'}
+    );
+
+  const client = new ShopperSearch({...config});
+  const responsePromise = () =>
+    client.productSearch({
+      parameters: {q: 'sony'},
+      headers: {authorization: 'Bearer test-auth'},
+    });
+
+  expect(responsePromise).rejects.toEqual(new Error('Error 400: Bad Request'));
```

Comment:
```suggestion
  await expect(responsePromise).rejects.toEqual(new Error('Error 400: Bad Request'));
```
Don't forget to await the expression so that it does what you expect it to!